java/DEVELOPER.md

```diff
@@ -0,0 +1,136 @@
+# Developer Guide
+
+This document describes how to set up your development environment to build and test the GLIDE for Redis Java wrapper.
+
+### Development Overview
+
+The GLIDE for Redis Java wrapper consists of both Java and Rust code. Rust bindings for the Java Native Interface are implemented using [jni-rs](https://github.com/jni-rs/jni-rs), and the Java JAR is built using [Gradle](https://github.com/gradle/gradle). The Java and Rust components communicate using the [protobuf](https://github.com/protocolbuffers/protobuf) protocol.
+
+### Build from source
+
+#### Prerequisites
+
+Software Dependencies
+
+-   git
+-   GCC
+-   pkg-config
+-   protoc (protobuf compiler) >= 26.1
+-   openssl
+-   openssl-dev
+-   rustup
+-   Java 11
+
+**Dependencies installation for Ubuntu**
+
+```bash
+sudo apt update -y
+sudo apt install -y openjdk-11-jdk git gcc pkg-config openssl libssl-dev unzip
+# Install rust
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+source "$HOME/.cargo/env"
+# Check that the Rust compiler is installed
+rustc --version
+# Install protobuf compiler
+PB_REL="https://github.com/protocolbuffers/protobuf/releases"
+curl -LO $PB_REL/download/v26.1/protoc-26.1-linux-x86_64.zip
+unzip protoc-26.1-linux-x86_64.zip -d $HOME/.local
+export PATH="$PATH:$HOME/.local/bin"
+# Check that the protobuf compiler is installed
+protoc --version
```

Comment:
Duplicate this block (`protoc`) to other systems or move it away from ubuntu section.